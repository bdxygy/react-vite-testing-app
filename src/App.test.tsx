import { render, screen, fireEvent } from '@testing-library/react';
import App from './App';

describe('App', () => {
  let component:  HTMLElement; 
  
  beforeEach(() => {
    jest.clearAllMocks();
    const {container} = render(<App />);
    component = container;
  });


  it('renders without crashing', async () => {
    // check to match snapshot
  expect(component).toMatchSnapshot();
  
  const h1 = screen.getByText('Vite + React');
   expect(h1).toBeInTheDocument();
  });

  it('should click the button increment', () => {
    const button = screen.getByRole('button');
    expect(button).toBeInTheDocument();

    fireEvent.click(button);

    expect(button).toHaveTextContent('count is 1');
    expect(component).toMatchSnapshot();
  });
});