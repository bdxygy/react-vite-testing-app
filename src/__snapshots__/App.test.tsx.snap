// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<div>
  <div>
    <a
      href="https://vitejs.dev"
      target="_blank"
    >
      <img
        alt="Vite logo"
        class="logo"
        src="test-file-stub"
      />
    </a>
    <a
      href="https://react.dev"
      target="_blank"
    >
      <img
        alt="React logo"
        class="logo react"
        src="test-file-stub"
      />
    </a>
  </div>
  <h1>
    Vite + React
  </h1>
  <div
    class="card"
  >
    <button>
      count is 
      0
    </button>
    <p>
      Edit 
      <code>
        src/App.tsx
      </code>
       and save to test HMR
    </p>
  </div>
  <p
    class="read-the-docs"
  >
    Click on the Vite and React logos to learn more
  </p>
</div>
`;

exports[`App should click the button increment 1`] = `
<div>
  <div>
    <a
      href="https://vitejs.dev"
      target="_blank"
    >
      <img
        alt="Vite logo"
        class="logo"
        src="test-file-stub"
      />
    </a>
    <a
      href="https://react.dev"
      target="_blank"
    >
      <img
        alt="React logo"
        class="logo react"
        src="test-file-stub"
      />
    </a>
  </div>
  <h1>
    Vite + React
  </h1>
  <div
    class="card"
  >
    <button>
      count is 
      1
    </button>
    <p>
      Edit 
      <code>
        src/App.tsx
      </code>
       and save to test HMR
    </p>
  </div>
  <p
    class="read-the-docs"
  >
    Click on the Vite and React logos to learn more
  </p>
</div>
`;
